Sources/Workspace/ResolverPrecomputationProvider.swift

```diff
@@ -121,23 +122,25 @@ private struct LocalPackageContainer: PackageContainer {
 
     func getDependencies(at version: Version, productFilter: ProductFilter) throws -> [PackageContainerConstraint] {
         // Because of the implementation of `reversedVersions`, we should only get the exact same version.
-        precondition(dependency?.checkoutState?.version == version)
```

Comment:
please double check logic still looks right

Reply:
Looks correct to me.